tests/platform/test_sfp.py

```diff
@@ -0,0 +1,103 @@
+"""
+Check SFP status and configure SFP
+
+This script covers test case 'Check SFP status and configure SFP' in the SONiC platform test plan:
+https://github.com/Azure/SONiC/blob/master/doc/pmon/sonic_platform_test_plan.md
+"""
+import logging
+import re
+import os
+import time
+
+from ansible_host import ansible_host
+
+
+def parse_presence(output_lines):
+    """
+    @summary: Parse the SFP presence information from command output
+    @param output_lines: Command output lines
+    @return: Returns result in a dictionary
+    """
+    res = {}
+    for line in output_lines:
+        fields = line.split()
+        if len(fields) != 2:
+            continue
+        res[fields[0]] = fields[1]
+    return res
+
+
+def parse_eeprom(output_lines):
+    """
+    @summary: Parse the SFP eeprom information from command output
+    @param output_lines: Command output lines
+    @return: Returns result in a dictionary
+    """
+    res = {}
+    for line in output_lines:
+        if re.match(r"^Ethernet\d+: .*", line):
+            fields = line.split(":")
+            res[fields[0]] = fields[1].strip()
+    return res
+
+
+def test_check_sfp_status_and_configure_sfp(localhost, ansible_adhoc, testbed):
```

Comment:
If i am not mistaken we also have low power mode option under sfputil. Should we add it as well?

Reply:
Sure. Added SFP low power mode testing in the new commit.